teamcity/PublishCSPForUnity.py

```diff
@@ -0,0 +1,239 @@
+    arg_parser.add_argument(
+        "--license",
+        help="Enter the license required for the Connected Spaces Platform package.",
+        default="Apache-2.0",
+    )
+    arg_parser.add_argument(
+        "--description",
+        help="Enter the description for the Connected Spaces Platform package.",
+        default="This package provides the binaries required to interface with the Connected Spaces Platform API.",
+    )
+    arg_parser.add_argument(
+        "--registry",
+        help="This is the upstream location of the package.",
+        default="https://npm.pkg.github.io/@magnopus-opensource",
+    )
+    arg_parser.add_argument(
+        "--repository",
+        help="This is the location of the Git repository.",
+        default="https://github.com/magnopus-opensource/connected-spaces-platform",
+    )
+    arg_parser.add_argument(
+        "--release-mode",
+        help="NPM release command, pack and publish are available.",
+        default="pack",
+    )
+
+    args = arg_parser.parse_args()
+
+    # Clean previously-generated package
+    shutil.rmtree(args.output_directory, ignore_errors=True)
+
+    os.makedirs(args.output_directory)
+
+    # Initialise Jinja environment
+    env = Environment(loader=FileSystemLoader(template_directory), extensions=["jinja2_workarounds.MultiLineInclude"])
+
+    minimal_meta_template = env.get_template("minimal.meta.jinja2")
+    source_meta_template = env.get_template("source.cs.meta.jinja2")
+    asmdef_template = env.get_template("asmdef.json.jinja2")
+    plugin_meta_template = env.get_template("plugin.meta.jinja2")
+    package_json_template = env.get_template("package.json.jinja2")
+
+    # Copy C# source code and generate meta files
+    source_output_directory = f"{args.output_directory}/Source"
+    os.mkdir(source_output_directory)
+
+    with open(f"{source_output_directory}.meta", "w") as f:
+        f.write(minimal_meta_template.render(guid=generate_guid(), importer_type="Default"))
+
+    for path in args.csharp_artifact_paths:
+        dir_util.copy_tree(path, source_output_directory)
+
+    for path, dirs, files in os.walk(source_output_directory):
+        for dir in dirs:
+            with open(f"{path}/{dir}.meta", "w") as f:
+                f.write(minimal_meta_template.render(guid=generate_guid(), importer_type="Default"))
+
+        for file in files:
+            with open(f"{path}/{file}.meta", "w") as f:
+                f.write(source_meta_template.render(guid=generate_guid()))
+
+    # Create C# assembly definition file and meta file
+    with open(f"{source_output_directory}/{args.assembly_name}.asmdef", "w") as f:
+        f.write(asmdef_template.render(assembly_name=args.assembly_name, root_namespace=args.assembly_namespace))
+
+    with open(f"{source_output_directory}/{args.assembly_name}.asmdef.meta", "w") as f:
+        f.write(
+            minimal_meta_template.render(guid="1678634e69642a55e208df4d4d895a52", importer_type="AssemblyDefinition")
```

Comment:
Can we add a comment here just to say what this guid is and that it should not be changed?